verticles/graphql/src/main/java/com/gentics/mesh/graphql/type/NodeTypeProvider.java

```diff
@@ -407,16 +407,29 @@ public Versioned<List<GraphQLFieldDefinition>> createNodeInterfaceFields(GraphQL
 				.argument(createNodeVersionArg())
 				.type(new GraphQLTypeReference(NODE_REFERENCE_PAGE_TYPE_NAME))
 				.dataFetcher(env -> {
+					GraphQLContext gc = env.getContext();
 					NodeContent content = env.getSource();
-					ContainerType type = getNodeVersion(env);
-					Stream<NodeReferenceIn> stream = NodeReferenceIn.fromContent(context, content, type);
-					Map<String, ?> filterInput = env.getArgument("filter");
-					if (filterInput != null) {
-						stream = stream.filter(nodeReferenceFilter(context).createPredicate(filterInput));
+					if (content == null) {
+						return null;
+					}
+					HibNode node = content.getNode();
+					if (node == null) {
+						return null;
 					}
+					List<String> languageTags = getLanguageArgument(env, content);
+					ContainerType type = getNodeVersion(env);
```

Comment:
This is not used anymore, which changes the behaviour.
In the original implementation, the type was take from the environment, the batch loader will now use the type of each content to load the referencing node contents.

Reply:
This can be tested with
`referencedBy(version: draft)` vs. `referencedBy(version: published)`

Reply:
Fixed.